runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -764,9 +765,8 @@ J9::Z::TreeEvaluator::inlineVectorizedStringIndexOf(TR::Node* node, TR::CodeGene
 
       // pattern header not found in first 16 bytes of the string
       // Load the next 16 bytes of the string and continue
-      generateRIEInstruction(cg, TR::InstOpCode::getCmpRegAndBranchRelOpCode(), node, stringIndexReg, maxIndexReg, labelPatternNotFound, TR::InstOpCode::COND_BNL);
       generateRRInstruction(cg, TR::InstOpCode::getAddRegOpCode(), node, stringIndexReg, loadLenReg);
-      generateRIEInstruction(cg, TR::InstOpCode::getCmpRegAndBranchRelOpCode(), node, stringIndexReg, stringLenReg, labelPatternNotFound, TR::InstOpCode::COND_BNL);
+      generateRIEInstruction(cg, TR::InstOpCode::getCmpRegAndBranchRelOpCode(), node, stringIndexReg, maxIndexReg, labelPatternNotFound, TR::InstOpCode::COND_BH);
```

Comment:
Might be worth adding an iComment to explain this comparison.  i.e. "Updated stringIndex for next iteration exceeds maxIndex of valid match.  Full pattern cannot be matched."

Reply:
Fixed in [force-push](https://github.com/eclipse-openj9/openj9/compare/b2389b0722853a5cac394cbb8948c6e8ebdc9cdc..c20200d).